httpclient-backend/zio/src/main/scala/sttp/client/httpclient/zio/ZioWebSocketHandler.scala

```diff
@@ -0,0 +1,36 @@
+package sttp.client.httpclient.zio
+
+import sttp.client.httpclient.WebSocketHandler
+import sttp.client.httpclient.internal.NativeWebSocketHandler
+import sttp.client.impl.zio.RIOMonadAsyncError
+import sttp.client.ws.{WebSocket, WebSocketEvent}
+import sttp.client.ws.internal.AsyncQueue
+import sttp.model.ws.WebSocketBufferFull
+import zio.{Queue, Runtime, Task, ZIO}
+
+object ZioWebSocketHandler {
+  private class ZioAsyncQueue[A](queue: Queue[A], runtime: Runtime[Any]) extends AsyncQueue[BlockingTask, A] {
+    override def offer(t: A): Unit = {
+      if (!runtime.unsafeRun(queue.offer(t))) {
+        throw new WebSocketBufferFull()
```

Comment:
maybe this should be in fact an `IllegalStateException` or sth similar indicating a bug - we shouldn't ever get into a situation where the buffer is full?

Reply:
That's true, the name of the exception isn't right